### PR TITLE
cgo,native: fix Dormqr work checks

### LINKED ISSUE #117: cgo: Dormqr comment not in sync with reality
The comment says:
```
// ... lwork == -1 will not run Dormqr but will instead write the minimum
// work necessary to work[0]. If len(work) < lwork, Dormqr will panic.
```
The code does:
```
        if lwork == -1 {
                 if left {
                         work[0] = float64(m)
                         return
                 }
                 work[0] = float64(n)
                 return
         }
 
         if left {
                 if lwork < n {
                         panic(badWork)
                 }
         } else {
                 if lwork < m {
                         panic(badWork)
                 }
         }
```
That is,
1. Values put into work[0] and checked values of lwork seem to be swapped.
2. len(work) < lwork is not checked.

### LINKED ISSUE #119: native: check whether len(work) >= lwork in Dormqr

### PR DESCRIPTION
**Pull request by kortschak**
Fixes #117.
Fixes #119.

@btracey @vladimir-ch Please take a look.

### CODE REVIEW

File: cgo/lapack.go [hunk 1]
```diff
@@ -951,13 +951,15 @@ func (impl Implementation) Dormqr(side blas.Side, trans blas.Transpose, m, n, k
...
-			work[0] = float64(m)
+			work[0] = float64(n)
 			return
 		}
-		work[0] = float64(n)
+		work[0] = float64(m)
 		return
 	}
-
+	if len(work) != lwork {
```
Comment (vladimir-ch) [inline cgo/lapack.go]: (At least at the moment) the rest of the code (and the doc comment) use `if len(work) < lwork`.

Comment (kortschak) [inline cgo/lapack.go]: Ugh. See comment about reading in other thread.

Done.

**Comment by vladimir-ch** [comment]
LGTM but perhaps we should wait for @btracey ?

**Comment by kortschak** [comment]
Absolutely. The whole row versus column thing gives me the heebie geebies.

**Comment by btracey** [comment]
LGTM

It's not so bad once you've implemented float64-BLAS and a few lapack functions (except for the times when it still is)

**Pull request merged**